### PR TITLE
Migrate `mock-observations` to `builtins`

### PR DESCRIPTION
**Pull request by jmcook1186**
### Types of changes

- [x] Enhancement (project structure, spelling, grammar, formatting)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### A description of the changes proposed in the Pull Request

Migrates `mock-observations` from plugins to builtins.

**Dev review** - @narekhovhannisyan please check that changes in the plugin source code are benign (specifially making execute synchronous).

Addresses #678

### CODE REVIEW

File: src/builtins/mock-observations/README.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      pipeline:
+        - mock-observations
+      inputs:
+```
+
+You can run this example `manifest` by saving it as `./examples/manifests/test/mock-observation.yml` and executing the following command from the project root:
+
+```sh
+npm i -g @grnsft/if
+npm i -g @grnsft/if-plugins
```
Comment (manushak) [inline src/builtins/mock-observations/README.md:94]: please, remove this row and update manifest path

File: src/builtins/mock-observations/README.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+description: example invoking mock-observation plugin
+tags:
+initialize:
+  outputs:
+    - yaml
+  plugins:
+    mock-observations:
+      kind: plugin
+      method: MockObservations
+      path: 'builtins'
```
Comment (manushak) [inline src/builtins/mock-observations/README.md:63]: should be `builtin`

Comment (jmcook1186) [reply src/builtins/mock-observations/README.md:63]:
```suggestion
      path: 'builtin'
```

File: src/builtins/mock-observations/README.md (hunk 1 above)
Comment (jmcook1186) [reply src/builtins/mock-observations/README.md:94]:
```suggestion
```

File: src/builtins/mock-observations/README.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+              max: 99
+tree:
+  children:
+    child:
+      pipeline:
+        - mock-observations
+      inputs:
+```
+
+You can run this example `manifest` by saving it as `./examples/manifests/test/mock-observation.yml` and executing the following command from the project root:
```
Comment (jmcook1186) [inline src/builtins/mock-observations/README.md:90]:
```suggestion
You can run this example `manifest` by saving it as `manifests/plugins/mock-observation.yml` and executing the following command from the project root:
```

**Review by manushak — APPROVED** [review]

File: src/builtins/mock-observations/helpers/common-generator.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      );
+    }
+
+    return copyObject(config);
+  };
+
+  /**
+   * Generates next value by copying the validated config.
+   */
+  const next = (_historical: Object[] | undefined): Object =>
```
Comment (narekhovhannisyan) [inline src/builtins/mock-observations/helpers/common-generator.ts:34]: please fix this warnings

**Comment by narekhovhannisyan** [comment]
warnings seems still there

**Comment by jmcook1186** [comment]
one warning is still there, but I'm not sure that the parameter can really be removed - can you take a look?

**Pull request merged**